NoBeardProject/src/main/java/machine/NoBeardMachine.java

```diff
@@ -56,15 +58,18 @@ public int getCurrentLine() {
     private final CallStack callStack;
     private final ProgramMemory programMemory;
     private final ControlUnit controlUnit;
+    private final List<Integer> breakpoints;
```

Comment:
When reflecting to do the adding and removing of breakpoints more dynamic (and not deleting all breakpoints and adding them again after each step) I would prefer to organize breakpoints via a TreeSet. This does exactly what you need. You need not check for double entries (because its a set) and it is ordered (because its a tree).